besu/src/main/java/org/hyperledger/besu/controller/BesuController.java

```diff
@@ -211,6 +212,13 @@ BesuControllerBuilder fromGenesisConfig(
         throw new IllegalArgumentException("Unknown consensus mechanism defined");
       }
 
+      if (configOptions.getTerminalTotalDifficulty().isPresent()
+          && configOptions.getCheckpointOptions().isValid()
+          && (UInt256.fromHexString(configOptions.getCheckpointOptions().getTotalDifficulty().get())
+              .greaterOrEqualThan(configOptions.getTerminalTotalDifficulty().get()))) {
+        return new MergeBesuControllerBuilder().genesisConfigFile(genesisConfig);
+      }
+
       // wrap with TransitionBesuControllerBuilder if we have a terminal total difficulty:
       if (configOptions.getTerminalTotalDifficulty().isPresent()) {
```

Comment:
nit: this if could become an if else as the post-merge and pre-merge are exclusive